core/builtin.py

```diff
@@ -496,11 +505,12 @@ def Pushd(argv, dir_stack):
   dest_dir = argv[0]
   try:
     os.chdir(dest_dir)
+    dir_stack.append(os.getcwd())
+    PrintDirStack(dir_stack)
```

Comment:
Put the things that can't raise OSError outside the except: block.

I think it's better to append `dest_dir` rather than calling os.getcwd() again.